### DISCUSSION TITLE
Paypal shipment error for digial products

### DISCUSSION THREAD

**Discussion by mohitkumar29** [swellstores/community#524] [Help] [answered]
Hey there, i am working on building a custom checkout for the storefront using swell-js library, 

when testing paypal, i am getting this shipment error, 
![image](https://github.com/swellstores/community/assets/104753963/1f548309-8b8b-4c9f-958f-c405a95ed7c7)

this is how i am rendering the paypal button 

```js 
  swell.payment.createElements({
  paypal: {
    elementId: 'paypal-button',
    onSuccess: () => {
      // called on paypal payment success
      submitOrder();
    },
    onError: error => {
      // called on paypal payment error
      console.error(error);
      errorHandler(error);
    }
  }
});
``` 

i checked the payment intent network call that's fired when i click the paypal button, it has shipping preference set to **GET_FROM_FILE** while in the swell hosted checkout it's **NO_SHIPPING**. how can i configure and fix this?, thanks. 

![image](https://github.com/swellstores/community/assets/104753963/5a075352-a43d-4024-a1ea-323be0f32816)

**Comment by vstudenichnik-insoft** [comment, marked as answer]
@mohitkumar29 you can configure the PayPal button this way:

```
const requireShipping = Boolean(cart.shipment_delivery);

swell.payment.createElements({
  paypal: {
    elementId: 'paypal-button',
    require: {
      shipping: requireShipping, // default: true
    },
    onSuccess: () => {
      // called on paypal payment success
      submitOrder();
    },
    onError: error => {
      // called on paypal payment error
      console.error(error);
      errorHandler(error);
    }
  }
});
```
